Lib/pdb.py

```diff
@@ -1531,8 +1532,7 @@ def _runmodule(self, module_name):
         import runpy
         mod_name, mod_spec, code = runpy._get_module_details(module_name)
         self.mainpyfile = self.canonic(code.co_filename)
-        import __main__
-        __main__.__dict__.clear()
+        __main__ = types.ModuleType(module_name, 'New __main__ module for pdb.')
```

Comment:
This won't interact correctly with the interpreter's `-i` switch, as that always drops you into the *original* `__main__` namespace, even if it's been rebound in `sys.modules` by the application code.